SConstruct

```diff
@@ -8,28 +8,30 @@ from lsst.sconsUtils.utils import libraryLoaderEnvironment
 from lsst.utils import getPackageDir
 from lsst.ci.hsc.validate import *
 
-from SCons import Action
-
-class CallAction(Action.FunctionAction):
-    def __init__(self, ident, func, **kw):
-        self.ident = ident
-        self.func = func
-        Action.FunctionAction.__init__(self, self.func, kw)
-
-    # This function returns what is hashed to generate the function signature.
-    # This signature is what scons uses to decide if the build function has
-    # changed, and thus the target should be rebuilt. Unfortunately functools.partial
-    # or lambda cause scons to generate a different signature on each run, and thus
-    # rebuild everything. This function is overloaded here to ensure that a consistent
-    # function signature is generated
-    def get_contents(self, *args, **kwargs):
-        return str(self.ident)
-
-def makeSCons(cls, root, *args, **kwargs):
-    instance = cls(root)
-    return CallAction(str(args) + str(kwargs), functools.partial(instance.scons, *args, **kwargs))
+from SCons.Script import SConscript
+SConscript(os.path.join(".", "bin.src", "SConscript"))
 
 env = Environment(ENV=os.environ)
+env["ENV"]["OMP_NUM_THREADS"] = "1"  # Disable threading; we're parallelising at a higher level
+
+def validate(cls, root, dataId=None, **kwargs):
+    """!Construct a command-line for validation
+
+    @param cls  Validation class to use
+    @param root  Data repo root directory
+    @param dataId  Data identifier dict, or None
+    @param kwargs  Additional key/value pairs to add to dataId
+    @return Command-line string to run validation
+    """
+    if dataId:
+        dataId = dataId.copy()
+        dataId.update(kwargs)
+    else:
+        assert len(kwargs) == 0  # There's no dataId to update
+    cmd = [getExecutable("ci_hsc", "validate.py"), cls.__name__, root,]
+    if dataId:
+        cmd += ["--id %s" % (" ".join("%s=%s" % (key, value) for key, value in dataId.iteritems()))]
```

Comment:
Our developer guide mandates the use of `iteritems()`.

But: in this case, we know there aren't going to be many items in the dataId dict, so it's unlikely that this is saving you anything. Plus, if you used `items()`, it would work directly in Python 3, with no syntax changes necessary. Just saying...!